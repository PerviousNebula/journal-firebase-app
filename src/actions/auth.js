import Swal from 'sweetalert2';

import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { uiFinishLoading, uiStartLoading } from './ui';
import { types } from '../types/types';
import { noteLogout } from './notes';

export const startLoginEmailPassword = (email, password) => {
    return (dispatch) => {
        dispatch(uiStartLoading());

        return firebase.auth().signInWithEmailAndPassword(email, password)
            .then(({ user }) =>
                dispatch(login(user.uid, user.displayName))
            ).catch(e => {
                console.error(e);
                Swal.fire('Error', e.message, 'error');
            }).finally(() =>
                dispatch(uiFinishLoading())
            );
    };
};

export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase.auth().signInWithPopup(googleAuthProvider)
            .then(({ user  }) => 
                dispatch(login(user.uid, user.displayName))
            );
    };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
    return (dispatch) => {
        firebase.auth().createUserWithEmailAndPassword(email, password)
            .then(async({ user  }) => {
                await user.updateProfile({ displayName: name });
                dispatch(login(user.uid, user.displayName))
            }).catch(e => {
                console.error(e);
                Swal.fire('Error', e.message, 'error');
            });
    };
};

export const login = (uid, displayName) => ({
    type: types.login,
    payload: {
        uid,
        displayName,
    },
});

export const startLogout = () => {
    return async(dispatch) => {
        await firebase.auth().signOut();
        dispatch(logout());
        dispatch(noteLogout());
    };
};

export const logout = () => ({
    type: types.logout
});
