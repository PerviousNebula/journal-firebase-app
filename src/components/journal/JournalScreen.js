import React from 'react';
import { useSelector } from 'react-redux';

import { NothingSelected } from './NothingSelected';
import { NoteScreen } from '../notes/NoteScreen';
import { Sidebar } from './Sidebar';

export const JournalScreen = () => {
    const { active } = useSelector(state => state.notes);

    return (
        <div
            className="journal__main-content animate__animated animate__fadeIn animate__faster"
        >
            <Sidebar />

            <main>
                {
                    active ? <NoteScreen /> : <NothingSelected />
                }
            </main>
        </div>
    )
}
