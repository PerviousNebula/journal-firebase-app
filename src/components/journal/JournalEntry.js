import React from 'react';
import moment from 'moment';
import { useDispatch } from 'react-redux';

import { activeNote } from '../../actions/notes';

export const JournalEntry = ({ id, date, title, body, url }) => {
    const dispatch = useDispatch();
    const noteDate = moment(date);

    const handleEntryClick = () => {
        dispatch(activeNote(id, {
            title,
            body,
            url,
            date,
        }));
    }

    return (
        <div
            className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
            onClick={ handleEntryClick }
        >
            {
                url &&
                <div
                    className="journal__entry-picture"
                    style={{
                        backgroundImage: `url(${ url })`,
                        backgroundSize: 'cover'
                    }}
                >
                </div>
            }
            <div className="journal__entry-body">
                <p className="journal__entry-title">{ title }</p>
                <p className="journal__entry-content">{ body }</p>
            </div>
            <div className="journal__entry-date-box">
                <span>{ noteDate.format('dddd') }</span>
                <h4>{ noteDate.format('Do') }</h4>
            </div>
        </div>
    )
}
