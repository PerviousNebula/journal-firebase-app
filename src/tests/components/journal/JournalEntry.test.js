import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import { activeNote } from '../../../actions/notes';

import { JournalEntry } from '../../../components/journal/JournalEntry';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
    id: '123',
    title: 'Hola',
    body: 'mundo',
    url: 'https://algunlugar.com/foto.jpg'
};
const wrapper = mount(
    <Provider store={ store }>
        <JournalEntry { ...note } />
    </Provider>
);

describe('Pruebas en <JournalEntry />', () => {

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de activar la nota', () => {
        wrapper.find('.journal__entry').prop('onClick')();

        expect(store.dispatch).toHaveBeenCalledWith(activeNote(note.id, { ...note }));
    });

});
