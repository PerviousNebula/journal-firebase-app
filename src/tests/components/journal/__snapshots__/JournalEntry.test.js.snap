// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalEntry /> debe mostrarse correctamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://algunlugar.com/foto.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Hola
    </p>
    <p
      className="journal__entry-content"
    >
      mundo
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Friday
    </span>
    <h4>
      3rd
    </h4>
  </div>
</div>
`;
